ScrapeRepos.ts

```diff
@@ -0,0 +1,356 @@
+  try {
+    const timeModes = ['daily', 'weekly', 'monthly'];
+    const response = await axios.get('https://github.com/trending?since=' + timeModes[timeMode]);
+    const $ = cheerio.load(response.data);
+    const repos: string[] = [];
+
+    $('h2 a').each((i: number, el: cheerio.Element) => {
+      const repoName = $(el).text().trim();
+      repos.push(repoName);
+    });
+    
+    var trendingSplit: string[] = [];
+    // trim the repos to be correctly formatted
+    repos.forEach((repo) => {
+      let trimmedName = repo.replace(/\n\s+/g, '');
+      trimmedName = trimmedName.replace(/\//g, "");
+      const stringSplit = trimmedName.split(" ");
+      trendingSplit.push(stringSplit[0]);
+      trendingSplit.push(stringSplit[1]);
+    });
+
+    return trendingSplit;
+  } catch (error) {
+    console.error(error);
+  }
+}
+
+
+/**  This function imports the ReadMe.md file for a repository (if it can be located)
+* @param {string} owner - owner of the repo
+* @param {string} name - name of the repo
+* @returns {string} a string containing the text of the repo. Returns null if the repo can't be located
+*/
+async function getReadme(owner: String, name: String) {
+  // these paths exists to check in multiple locations for the readme files
+  // these can be constantly expanded if new paths are being found
+  const readmePaths = [`https://raw.githubusercontent.com/${owner}/${name}/release/readme.md`,
+                          `https://raw.githubusercontent.com/${owner}/${name}/dev/README.rst`,
+                        `https://raw.githubusercontent.com/${owner}/${name}/main/README.md`,
+                        `https://raw.githubusercontent.com/${owner}/${name}/master/README.md`];
+  
+  for(let i = 0; i < readmePaths.length; i++){
+    try{
+      const response = await axios.get(readmePaths[i])
+      const content = response.data
+      const converter = new showdown.Converter();
+
+      // Use the converter object to convert Markdown to HTML to String:
+      const html = converter.makeHtml(content);
+      const html2 = html.toString();
+      const text = html2.replace(/<[^>]*>/g, '');
+      const text2 = text.replace(/\n\s+/g, '');
+      const text3 = text2.replace(/\//g, "");
+      return text3;
+    }
+    catch(error){
+      
+    }
+  }
+  console.log("ReadMe couldn't be found")
+  return null
+}
+  
+
+/** Gets the repo's information via GitHub's GraphQL API
+ * @param {string} query GraphQL query for the repo (including owner and name)
+ * @param {string} authToken personal authorization token
+ * @returns {any[]} the json data for the requested repo as by the graphql query; null on error
+ */
+async function getRepoInfo(query: String, authToken: String){
+  try{
+    const response = await  axios.post('https://api.github.com/graphql', {
+      query: query
+    }, {
+      headers: {
+        'Authorization': authToken 
+      }
+    });
+    const output = response.data.data.repository;
```

Comment:
Please return response.data.data.repository directly. Don't need to create a variable.